Bundle/CriteriaBundle/DataSource/RequestDataSource.php

```diff
@@ -40,7 +40,11 @@ public function getLocaleFormParams()
         return [
             'type'    => ChoiceType::class,
             'options' => [
-                'choices' => $this->availableLocales,
+                'choices'           => $this->availableLocales,
+                'choices_as_values' => true,
+                'choice_label'      => function ($value) {
```

Comment:
When e.g `$this->availableLocales = ['fr', 'en', 'es']`, the values will be equals to 0, 1, 2. To avoid that, choices_as_values inverts as ['fr' => 0, 'en' => 1, 'es' => 2]. 
The choice_label option converts it to ['fr' => 'fr', 'en' => 'en', 'es' => 'es'] as espected